src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -1096,14 +1153,19 @@ private void DecodeParameterOrThrow(ref BlobReader signatureReader, /*out*/ ref
             bool refersToNoPiaLocalType;
 
             SignatureTypeCode typeCode;
-            info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode);
+            info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode, IsAcceptedIsConstModifierType, out bool isConstFound);
```

Comment:
This call allocates a delegate instance each time. Consider either passing in a value that represents the allowed custom modifier type (say `enum AllowedModifierType { None, Volatile, IsConst }`), or pass in a lambda with no closure (a `Func<MetadataDecoder, TypeSymbol, bool>` arg: `(decoder, type) => decoder.IsAcceptedIsConstModifierType(type)`).